lib/tasks/errors.rake

```diff
@@ -0,0 +1,31 @@
+namespace :errors do
+  desc <<-END_DESC
+Fetches stacktrace for given request id
+
+This task is useful to get stacktrace of error from logs in log directory. The request is identified by given request_id that is required.
+You can also specify log file where request logs has been saved. Default option is 'production.log'.
+
+Available conditions:
+  * request_id = Request id from 500 page
+  * log_file = specify another log file to lookup (default is 'production.log')
+
+Examples:
+  # foreman-rake errors:fetch_stacktrace request_id=aabbccss5 [log_file=development.log]
+  END_DESC
+
+  task :fetch_stacktrace do
+    request_id = ENV['request_id']
+    logfile = ENV['log_file'] || 'production.log' # default file where we want to find is production.log
+    unless request_id
+      puts "Can't find log without request_id"
+      next
+    end
+    file_path = File.join(Foreman::Logging.log_directory, logfile)
+    result = `grep #{request_id} #{file_path}`
```

Comment:
Not sure how relevant, but I'm always a fan of quoting arguments. open3 also has methods to pass in an array of arguments. Probably also worth to think about stderr handling.

Reply:
quoting is a good idea just to be safe, open3 is maybe overkill but no strong opinion on this

Reply:
I think I will stick with this solution. It's readable and It's not a big deal.

Reply:
I'd at least quote the arguments.

Reply:
this is still valid